beacon-chain/db/kv/blocks.go

```diff
@@ -415,20 +418,18 @@ func fetchBlockRootsBySlotRange(
 	}
 	min := bytesutil.Uint64ToBytesBigEndian(startSlot)
 	max := bytesutil.Uint64ToBytesBigEndian(endSlot)
-	var conditional func(key, max []byte) bool
-	if endSlot == 0 {
-		conditional = func(key, max []byte) bool {
-			return key != nil
-		}
-	} else {
-		conditional = func(key, max []byte) bool {
-			return key != nil && bytes.Compare(key, max) <= 0
-		}
+
+	conditional := func(key, max []byte) bool {
+		return key != nil && bytes.Compare(key, max) <= 0
 	}
-	rootsRange := (endSlot - startSlot) / step
 	if endSlot < startSlot {
-		rootsRange = 0
+		return nil, errInvalidSlotRange
+	}
+	// Return nothing with an end slot of 0.
+	if endSlot == 0 {
+		return [][]byte{}, nil
```

Comment:
Shouldn't this return the genesis block?

Reply:
We always retrieve the genesis block by its root and not by slots to overcome the limitations of zero values in Go throughout the codebase

Reply:
Perhaps worth adding as a comment @nisdas